### ISSUE TITLE
Plans for Gitea 1.22

### ISSUE THREAD

**Issue by delvh** [go-gitea/gitea#27194]
Hi all,

Gitea `v1.21.0-rc0` has been released.
We have created the `release/v1.21` branch accompanied by the `v1.22.0-dev` tag and are now using `main` for `1.22` development.
As always, this issue is intended to show what you can expect from the next version.

## Schedule

Gitea `1.22` is currently set to enter feature freeze on <time datetime="2023-12-17 23:59+00:00">2023-12-17 23:59 UTC</time>, so at the end of 2023.
This deadline is at the moment only a rough estimate and might change in the future.

<details><summary><h2>Previous Plans</h2></summary>

- #25123
- #23072
- #21598 
- #20022
- #18345
- #16429
- #14477
</details>

## :warning: Important

This summary is for things you **plan to implement yourself**, not things you want (from other people).
Wishes and other offtopic comments will be hidden.

**Comment by delvh** [comment]
I'll start with my plans:
- [x] Reviewing  **a lot** of PRs ([as always](https://github.com/go-gitea/gitea/pulls?q=is%3Apr+reviewed-by%3Adelvh+milestone%3A1.21.0+))
- [ ] Refactorings. Many of them. There are still way too many refactorings that need to be done at one point.
- [ ] Improve the documentation

So, overall, it will be a fairly feature free release from my side.
However, I'll gladly review features to get them merged.
Current contenders for it are
- #24257
- #25322
- #25439

**Comment by denyskon** [comment]
I'll try to review some of the open & forgotten PRs, specially:

- #25821
- #20620
- #24872
- #21221

For my own work, I plan to implement details pages for organisations and repo list and reverting of multiple commits.

Also of course as always, mobile vievport fixes and refactorings :)

I thought of taking a look at ways to switch to a new translation file format and cleaning up translations, but I don't know if I'll have time for that.

**Comment by silverwind** [comment]
- Introduce either unocss or tailwindcss
- Remove more fomantic
- Finish the new commit list (https://github.com/go-gitea/gitea/pull/24423)

**Comment by sahinakkaya** [comment]
At the start of this month, I realized due date for v21 milestone is end of month and I thought maybe I can find some time to finish #25439. Today I saw that my PR is moved to v22 milestone. Is there any chance that we finalize it before end of month and have it merged? @delvh Would it worth it if I rush? :)

**Comment by delvh** [comment]
No, we've been in feature freeze for 1.21 for a month now.
I'll try to get it into 1.22, but 1.21 is absolutely impossible now.

**Comment by lunny** [comment]
- [ ] Fix bugs
- [ ] Improve Actions
- [ ] Possible #18165
- [ ] Renderers Framework based on Docker/GRpc
- [ ] #25071

**Comment by Hypnotist1148** [comment]
Hey, how about global Teams instead of only Organization Teams?
Would be the first step to support LDAP group importing in the future, without the need to manually map a LDAP group to a specific Team in an Organization, would be an absolute Gitlab killer feature for corporate environments.

**Comment by denyskon** [comment]
@Hypnotist1148 Please open an Issue (or look if there already is one). This Issue is only for communication about what you are going to do.

**Comment by caixiangibm** [comment]
hi, when will the final forma gitea:1.21.0 image to be released in docker hub?
Is it possible for it to use go1.21.3 (released 2023-10-10) includes a security fix to the net/http package?

**Comment by delvh** [comment]
1. Wrong issue to discuss that.
2. The docker image uses the newest go 1.21 at the time of pulling
3. Not before we haven't released 1.21.0. Current plan is somewhere between the weekend and next week.

**Comment by viceice** [comment]
Is v1.22 already freezed? I've a small improvement for the nuget api with integration test.

- [ ] #28921

**Comment by denyskon** [comment]
@viceice No, we currently schedule the feature freeze for mid February. With some luck your PR should be merged until then :)

**Comment by zokkis** [comment]
Is it necessary to suggest changes for the new version here?
If yes, I would like to see #29122 in the new Release

**Comment by delvh** [comment]
1. No, it's not necessary
2. I am trying to get both of your recent PRs merged into 1.22 (especially #29117 since that one is an essential UI improvement). However, we need a second approver as well.

**Comment by sahinakkaya** [comment]
Is there a chance these two PR to be included in 1.22? #29191, #29210 
I think they'll do good with Contributors graph.

**Comment by sameer-m-dev** [comment]
Hi, not sure if the v1.22 has been frozen or not, but by any chance do we also have a possibility to merge 
https://github.com/go-gitea/gitea/pull/26673, this seems like a good change to have to add more customizability to Actions. Would love to help if any is required to get this merged

**Comment by delvh** [comment]
Ah, almost forgot to announce it here:

**Comment by Snowiiii** [comment]
> Gitea will enter feature freeze for 1.22 on Sunday, March 3, 2024 23:59 UTC. We aim to release 1.22.0-rc0 shortly after.
> 
> Major improvements already include :
> 
>     * Child PRs get retargeted once the parent PR has been merged
> 
>     * Contributors Graph of a repo
> 
>     * Files show the latest commit changing this file
> 
>     * Commits prefixed with `[skip ci]` are not run in Gitea Actions
> 
>     * Support for SHA256 repos
> 
>     * New merge style: `fast-forward only`
> 
>     * new and improved markdown syntax for writing alerts (`Tip`, `Note`, `Warning`, …)
> 
> 
> Until then, everything that is merged will be included in 1.22

Can we may remove JQuery before the 1.22 release ?

**Comment by denyskon** [comment]
@Snowiiii Probably not as Fomantic components depend on it

**Comment by Snowiiii** [comment]
> @Snowiiii Probably not as Fomantic components depend on it

So i guess we want to remove Fomantic ?
How long this would approximately take?

**Comment by delvh** [comment]
~2-3 years?

**Comment by theoludwig** [comment]
Why this issue have been unpinned? @silverwind 
Also what is the state of v1.22? Still no Release candidate?
It seems like the v1.23 milestone has already started: https://github.com/go-gitea/gitea/milestone/159, but from what I see, there are no separations of branch between v1.22 and v1.23? All PRs land on `main` branch.
Whereas usually, if I understand correctly, you create a `release/v1.22` from `main`, and then from there `main` becomes v1.23, and v1.22 is in feature freeze, but it seems like, it is not the case for this release?

**Comment by silverwind** [comment]
> Why this issue have been unpinned

Because I think pinned issues would better be reserved for "common issues/bugs", so people don't open so many duplicate issues, which was a problem in the last couple of weeks.

**Comment by techknowlogick** [comment]
@theoludwig the branch is created when the first rc is cut.

**Comment by tulzke** [comment]
When is the first rc planned?

**Comment by techknowlogick** [comment]
@tulzke rc0 was released a few hours ago.

**Comment by techknowlogick** [comment]
As rc0 of 1.22 is now out I will now close/lock this thread. Plans for 1.23 can be listed in the new thread (once it is created)
